test/unit/org/apache/cassandra/service/accord/serializers/JournalKeySerializerTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.accord.serializers;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import accord.local.Node;
+import accord.primitives.Routable;
+import accord.primitives.Timestamp;
+import accord.primitives.Txn;
+import accord.primitives.TxnId;
+import org.apache.cassandra.ServerTestUtils;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.harry.dsl.TestRunner;
+import org.apache.cassandra.harry.gen.Generator;
+import org.apache.cassandra.harry.gen.Generators;
+import org.apache.cassandra.service.accord.AccordKeyspace;
+import org.apache.cassandra.service.accord.JournalKey;
+
+public class JournalKeySerializerTest
+{
+    @BeforeClass
+    public static void setUp()
+    {
+        DatabaseDescriptor.daemonInitialization();
+        ServerTestUtils.prepareServer();
+    }
+
+    @Test
+    public void testOrder() throws Throwable
+    {
+        Node.Id node = new Node.Id(1);
+        Generator<Txn.Kind> kindGen = Generators.enumValues(Txn.Kind.class);
+        Generator<Routable.Domain> domainGen = Generators.enumValues(Routable.Domain.class);
+        Generator<JournalKey.Type> keyTypeGen = Generators.enumValues(JournalKey.Type.class);
+        Generator<JournalKey> keyGen = rng -> {
+            TxnId txnId = new TxnId(rng.nextLong(0, Timestamp.MAX_EPOCH),
```

Comment:
shouldn't `max_epoch` be included?  `nextLong` is exclusive no?  Its not documented in the interface but looks to be exclusive

Reply:
not a blocking comment, just an FYI

`accord.utils.AccordGens#txnIds()`

We maintain a generator for most accord types, and they are updated w/e we make changes

Reply:
I've double checked that the logic complies with `AccordGens`.

Reply:
Made it so that `max_epoch` is included; but looks like, for example, `AccordGens` generates those as `between(0, Timestamp.MAX_EPOCH)`. I've added +1 to gen in this patch as you suggested.